sopel/cli/run.py

```diff
@@ -113,17 +113,21 @@ def signal_handler(sig, frame):
 
 def add_legacy_options(parser):
     parser.add_argument("-d", '--fork', action="store_true",
-                        dest="daemonize", help="Daemonize Sopel")
+                        dest="daemonize",
+                        help=(
+                            "Daemonize Sopel. "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use ``sopel start -d`` instead)"))
```

Comment:
I thought the deprecation note was left off of this intentionally. https://github.com/sopel-irc/sopel/pull/1493/files#r262470506

Reply:
I don't remember exactly what I did. What I remember is that, from the moment `sopel start` exists and is available in master, `sopel -d` should be deprecated.

Reply:
Back in that PR, [I asked](https://github.com/sopel-irc/sopel/pull/1493/files#r262285696) about `-d`, "I'm guessing this has no deprecation note because e.g. sopel -d will work identically in Sopel 8, when start becomes the default action?" You replied, "Exactly!"

Unless that's no longer the case, this deprecation warning should be removed. (If it is no longer the case, you should open another PR to add the deprecation warning at runtime like the others.)

Reply:
Ha! Silly me.